### ISSUE TITLE
Weird behavior with the mask

### ISSUE THREAD

**Issue by CavalcanteLeo** [formkit/formkit#940]
### Reproduction

https://formkit.link/033713494457849aac57cb4ba02d9ae1

### Describe the bug

Just add few number values, then delete these values.

The cursor jumps around when adding and deleting

### Environment

• OS: [all]
• Browser [ chrome ]

**Comment by justin-schroeder** [comment]
Yep, looks like something we can take a look at — crux of the issue here is the suffix being used with a repeatable group — something we need to sort out.

**Comment by justin-schroeder** [comment]
Confirmed this is still reproducible on the Pro release branch when a reverse repeat mask uses a suffix. The suffix was being left in the mask runtime buffer after the pre-mutation was reversed, which caused alternating dropped digits/caret jumps.

Opened a Pro fix here: https://github.com/formkit/pro/pull/49

Local verification:
- `pnpm exec vitest run src/__tests__/maskInput.spec.ts -t "#940" --reporter verbose`
- `pnpm exec vitest run src/__tests__/maskInput.spec.ts src/__tests__/mask.spec.ts --reporter verbose`
- `pnpm typecheck:src`

Security review: this is client-side mask/caret state handling only; I did not find a security-sensitive trust boundary or injection issue.

**Comment by justin-schroeder** [comment]
This should be fixed in `@formkit/pro@0.130.0`, released today alongside `@formkit/*` 2.1.0. Please upgrade and let us know if anything still looks off. Thanks!
